### PR TITLE
parse json logs, duplicate record to message

### PR DESCRIPTION
**Pull request by schehata**
created a new field called message which will always contain the original string that is in record field. The record field is parsed and converted to an object.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -91,12 +91,27 @@ func httpHandler(ax *flusher.Axiom, runtimeDone chan struct{}) http.HandlerFunc
 			e["_time"], e["time"] = e["time"], nil
 
 			if e["type"] == "function" {
-				// parse the record
-				matches := logLineRgx.FindStringSubmatch(e["record"].(string))
-				if len(matches) == 5 {
-					e["message"] = matches[4]
-					e["record"] = map[string]any{"requestId": matches[2], "message": e["record"], "timestamp": matches[1]}
-					e["level"] = strings.ToLower(matches[3])
+				e["message"] = e["record"]
```
Comment (a-khaledf) [inline server/server.go:94]: wouldn't that mean, you are going to have message as unparsed string and record as parsed string.
that will duplicate the ingest data size for each user

**Review by a-khaledf — APPROVED** [review]
LGTM

File: server/server.go (hunk 1 above)
Comment (schehata) [reply server/server.go:94]: that's right, we wanted to always keep the original record string as source of truth. unfortunately we couldn't find a better way. we need a field that is always present for regex checking in dashboards or alerts, that's why we keep the text in `message` field.

File: server/server.go [hunk 2]
```diff
@@ -91,12 +91,27 @@ func httpHandler(ax *flusher.Axiom, runtimeDone chan struct{}) http.HandlerFunc
...
+				if recordStr, ok := e["record"].(string); ok {
+					recordStr = strings.Trim(recordStr, "\n")
+					// parse the record
+					// first check if the record is a json object, if not parse it as a text log line
+					if recordStr[0] == '{' && recordStr[len(recordStr)-1] == '}' {
```
Comment (bahlo) [inline server/server.go:99]: This panics if the string is empty, please check first

**Review by bahlo — APPROVED** [review]
Approving, but please add the bounds check before merging

File: server/server.go (hunk 2 above)
Comment (schehata) [reply server/server.go:99]: great catch. fixed.

**Pull request merged**